library/core/src/cmp.rs

```diff
@@ -27,16 +30,21 @@ pub(crate) use bytewise::BytewiseEq;
 
 use self::Ordering::*;
 
-/// Trait for equality comparisons.
+/// Trait for comparisons using the equality operator.
+///
+/// Implementing this trait for types provides the `==` and `!=` operators for
+/// those types.
 ///
 /// `x.eq(y)` can also be written `x == y`, and `x.ne(y)` can be written `x != y`.
 /// We use the easier-to-read infix notation in the remainder of this documentation.
 ///
-/// This trait allows for partial equality, for types that do not have a full
-/// equivalence relation. For example, in floating point numbers `NaN != NaN`,
-/// so floating point types implement `PartialEq` but not [`trait@Eq`].
-/// Formally speaking, when `Rhs == Self`, this trait corresponds to a [partial equivalence
-/// relation](https://en.wikipedia.org/wiki/Partial_equivalence_relation).
+/// This trait allows for comparisons using the equality operator, for types
+/// that do not have a full equivalence relation. For example, in floating point
+/// numbers `NaN != NaN`, so floating point types implement `PartialEq` but not
+/// [`trait@Eq`]. Formally speaking, when `Rhs == Self`, this trait corresponds
+/// to a [partial equivalence relation].
```

Comment:
Reading this, I had the concerns that Ralf currently expressed.

PartialEq is slightly more than a partial equivalence relation per se, precisely because we do not usually discuss equivalence relations between different types. In the non-Self case, it is _also_ the programmer's understanding of how the two types may be mapped into a larger space that contains the values of both types, which then contains equivalence relations and equivalence classes (or, in some cases, only partially so). I believe trying to express this by negation is part of why this documentation is so confused.

Obviously, in the simplest case, the mapping is the identity.

You do not need to address all of this problem, it is more a thought I had on how it might be improved.

Reply:
Honestly, thinking about it, it's the case even in the Self case, but there we can declare this as "working as intended": the interpretation operation, there, is the one intrinsic to the type.